### PR TITLE
Recent searches text matching

### PR DESCRIPTION
**Pull request by nmanu1**
Filter recent searches by whether they are Levenshtein matches for the input query text. Also, updated so that entity previews are displayed when focusing on a recent search.

J=SLAP-1802
TEST=manual

Check that recent searches are properly filtered when typing and entity previews are displayed on focus.

### CODE REVIEW

File: src/components/VisualAutocomplete/VisualSearchBar.tsx [hunk 1]
```diff
@@ -90,8 +91,12 @@ export default function VisualSearchBar({
       clearRecentSearches();
     }
   }, [clearRecentSearches, hideRecentSearches])
-  const haveRecentSearches = !hideRecentSearches && recentSearches?.length !== 0;
-  
+
+  const [filteredRecentSearches, setFilteredRecentSearches] = useState(recentSearches);
+  useEffect(() =>
```
Comment (yen-tt) [inline src/components/VisualAutocomplete/VisualSearchBar.tsx:96]: why is this useEffect call needed? I think we only need to call `setFilteredRecentSearches` when onInputFocus (user click or type on searchbar) event is trigger.

Comment (nmanu1) [reply src/components/VisualAutocomplete/VisualSearchBar.tsx:96]: I had added this to handle recent searches changing when the dropdown is open (like if recent searches were hidden/cleared). but this seems unlikely now that I think about it more, so I'll remove it

**Review by yen-tt — APPROVED** [review]

File: src/components/VisualAutocomplete/VisualSearchBar.tsx [hunk 2]
```diff
@@ -211,6 +214,9 @@ export default function VisualSearchBar({
         }}
         onInputFocus={value => {
           updateEntityPreviews(value);
+          setFilteredRecentSearches(recentSearches?.filter(search =>
```
Comment (cea2aj) [inline src/components/VisualAutocomplete/VisualSearchBar.tsx:217]: what's the reason for calling this here instead of `onInputChange`?

Comment (nmanu1) [reply src/components/VisualAutocomplete/VisualSearchBar.tsx:217]: calling it `onInputFocus` includes the events (i.e. typing) that trigger `onInputChange` as well as triggering when clicking on the input bar. for example, this is needed to filter the recent searches when initially clicking on the input bar (with or without text already present)

File: src/components/VisualAutocomplete/VisualSearchBar.tsx [hunk 3]
```diff
@@ -211,6 +214,9 @@ export default function VisualSearchBar({
         }}
         onInputFocus={value => {
           updateEntityPreviews(value);
+          setFilteredRecentSearches(recentSearches?.filter(search => 
+            answersUtilities.isCloseMatch(search.query, value)
```
Comment (cea2aj) [inline src/components/VisualAutocomplete/VisualSearchBar.tsx:218]: If someone clicks on the search bar but hasn't typed anything, do we want to show a certain number of recent searches without running the `isCloseMatch` function? Has UX given us any guidance there?

Comment (nmanu1) [reply src/components/VisualAutocomplete/VisualSearchBar.tsx:218]: By default, we've been showing up to 5 recent searches when there is no input text, which would also be the case here (all matches would be true for empty search string). I'm not sure if we chose that number or if we got that from UX.

Comment (cea2aj) [reply src/components/VisualAutocomplete/VisualSearchBar.tsx:218]: alright, sounds good

**Review by cea2aj — APPROVED** [review]

**Pull request merged**